### ISSUE TITLE
How to use Noise in Dropout?

### ISSUE THREAD

**Issue by anpki** [tflearn/tflearn#1021]
Hi all,

so I'm trying to train a CNN to recognise emotions (6 currently) from audio data and since my database is somewhat small, I want to improve the validation/prediction performance of my network. 
I read about the possibility to add noise to the data when using TFLearn's dropout function, but I'm not really understanding the Docs here. 
Let's assume my fully connected layer is like this:

`fc2 = fully_connected(fc1, 1024, activation='relu')`
` fc2 = dropout(fc2, keep_rate, noise_shape=[ X , 1024])`

What do I have to fill in for the X? I tried the batch size, but that's not working.. (and I'm even not sure if it is correct at all). Has anyone an idea/solution?

Secondly, I just have a question concerning the validation and prediction results. My network has a relatively good training performance and not so good validation performance (it's stuck around ~50%) when I train it on fewer emotions. But when I use the predict function to see what it thinks about new files, the performance is really good. So, how does it come that the validation is bad, but training + prediction is good? Do I ignore the validation and rely on the prediction, or should I stop caring about the prediction and try to push the validation? (And then: how? I tried changing batch size, dropout threshold, number of filter/units per layer, activation functions, learning rate, number of hidden layers .. )
